core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -1867,14 +1879,17 @@ copy_buffer_to_iovec_app(wasm_module_inst_t module_inst, uint8 *buf_begin,
         }
 
         if (buf >= buf_begin + buf_size
-            || buf + data->buf_len < buf /* integer overflow */
-            || buf + data->buf_len > buf_begin + buf_size) {
+            || buf + data->buf_len < buf /* integer overflow */) {
             break;
         }
 
+        // If our app buffer size is smaller than the amount to be copied,
+        // only copy the amount in the app buffer
```

Comment:
Could you use the `/* .. */` to add comments? We usually use it in the C source code.

Reply:
And wandering what the PR title means? Does it mean that we copy no more than the max size of app buffer from sender, but not the size that receiver hopes to receive?

Reply:
Will address thanks. I'll rename the PR. This will the received bytes into the app buffer, although it's a larger size. Eg with the TCP client example, the app buffer is 1024 bytes. However, the received bytes are 20 or so, so we will copy 20 bytes into the app buffer rather than the full 1024 bytes.